src/components/Pages/MainPage/MainPage.tsx

```diff
@@ -22,16 +22,16 @@ import { Props } from './MainPage.types'
 import styles from './MainPage.module.css'
 
 function MainPage(props: Props) {
-  const { isLoading, profileAddress, loggedInAddress, isBlocked, isCreationsTabEnabled, isAssetsTabEnabled } = props
+  const { isLoading, profileAddress, loggedInAddress, isBlocked, isCreationsTabEnabled, isAssetsTabEnabled, isLoadingFeatures } = props
   const view = getView(loggedInAddress, profileAddress)
   const isMobile = useMobileMediaQuery()
   const navigate = useNavigate()
   const params = useParams()
 
   if (
     !isTabValid(params.tab) ||
-    (params.tab === AccountTabs.ASSETS && !isAssetsTabEnabled) ||
-    (params.tab === AccountTabs.CREATIONS && !isCreationsTabEnabled)
+    (params.tab === AccountTabs.ASSETS && !isAssetsTabEnabled && !isLoadingFeatures) ||
```

Comment:
`isAssetsTabEnabled` and `isCreationsTabEnabled` come from a selector that already checks if the features were initially loaded, would you mind explaining to me why is this flag needed?

Reply:
Yesss! When refreshing the page in any of the tabs its doing this check before the actual features are fetch so its automatically redirecting to the main page